data/txHandler.go

```diff
@@ -564,10 +570,11 @@ func (handler *TxHandler) dedupCanonical(ntx int, unverifiedTxGroup []transactio
 func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) network.OutgoingMessage {
 	var msgKey *crypto.Digest
 	var isDup bool
+	var peers map[network.Peer]struct{}
 	if handler.msgCache != nil {
 		// check for duplicate messages
 		// this helps against relaying duplicates
-		if msgKey, isDup = handler.msgCache.CheckAndPut(rawmsg.Data); isDup {
+		if msgKey, peers, isDup = handler.msgCache.CheckAndPut(rawmsg.Data, rawmsg.Sender); isDup {
```

Comment:
I'm probably missing something but I'm not sure how this is meant to work. 

- when first copy of the transaction T comes through `peers` here will be a copy of the peers value from CheckAndPut and peers set will have the single peer A. This copy of peers gets added onto the queue
- second copy of transaction T comes from peer B . It will create a copy of the existing map and add B to it  and update the value in the cache correctly.
- when we Remember T from the message put on the backlog by A the `peers` value we will Relay it but `wi.peers` will still only contain peer A since it had it's copy from the first pass.

Reply:
oh, you are absolutely right. It worked before I introduced map copies since they were accessed in wsnetwork latter and caused raises. Good catch, working on a test and will fix this.

Reply:
Switched to *sync.Map since it appears to be an ideal use